jenkinsapi/job.py

```diff
@@ -64,7 +64,17 @@ def __init__(self, url, name, jenkins_obj):
             'hg': self._get_hg_branch,
             None: lambda element_tree: []
         }
-        self.url = url
+
+        if jenkins_obj.base_server_url() and jenkins_obj.get_use_baseurl():
+            job_hostname = urlparse.urlparse(url).netloc
+            jenkins_obj_hostname = urlparse.urlparse(jenkins_obj.base_server_url()).netloc
+            if jenkins_obj_hostname and job_hostname != jenkins_obj_hostname:
+                self.url = url.replace(job_hostname, jenkins_obj_hostname)
+            else:
```

Comment:
Is this `else` statement necessary here?

Reply:
I've re-written the logic so we don't have this else.